File: projects/m2/023-scrabble-score/solutions/jimmyhu/solution-scrabble-score.py
def score(word):
    values = {
        '1':['A', 'E', 'I', 'L', 'N', 'O', 'R', 'S', 'T', 'U'],
        '2':['D', 'G'],
        '3':['B', 'C', 'M', 'P'],
        '4':['F', 'H', 'V', 'W', 'Y'],
        '5':['K'],
        '8':['J', 'X'],
        '10':['Q','Z'],
    }
    points = 0
    for letter in word:
        for val in values:
            if letter in values[val]:
                points += int(val)
    return points
def main():
    wrd = input('Enter the word: ')
    print(f"This word score is: {score(wrd.upper())}")
if __name__ == '__main__':
    main()

File: projects/m2/023-scrabble-score/valentinaBazzacchi/solution-scrabble-score.js
// MAIN PROGRAM
const lettersArr = [
    ["A", "E", "I", "L", "N", "O", "R", "S", "T", "U"], ["D", "G"],
    ["B", "C", "M", "P"], ["F", "H", "V", "W", "Y"], ["K"], ["J", "X"], ["Q", "Z"]
];

let lettersPoints = new Map();
lettersPoints = createMap(lettersArr);

let word = prompt("Enter a word");
alert(`The Scrabble score for this word is: ${scrabbleScore(word, lettersPoints)}`);

// // Display map --> key: array of letters, value: points (int)
// for (let entry of lettersPoints.entries()) {
//     console.log(entry);
// }

// FUNCTIONS
function createMap(arr) {
    for (let y = 0, i = 1; y < arr.length; y++) {
        if (i === 6) {
            lettersPoints.set(arr[y], i + 2);
        } else if (i === 7) {
            lettersPoints.set(arr[y], i + 3);
        } else {
            lettersPoints.set(arr[y], i);
        }
        i++;
    } // end first for loop
    return lettersPoints;
} // end function

function scrabbleScore(str, map) {
    let totalPoints = 0;
    for (let letter of str.toUpperCase()) {
        map.forEach((point, arr) => {
            arr.forEach((elem) => {
                if (letter === elem) {
                    totalPoints += point;
                } // end condition
            }); // end arr.for.Each
        }); // end map.for.Each
    } //end for...of
    return totalPoints;
} // end function


